tests/pure_pursuit_tests.cpp

```diff
@@ -116,4 +123,41 @@ TEST_CASE( "Test get point on path", "[point_path]" )
         auto point = test_class.get_point_on_path_test( -15 );
         CHECK( point == Point3D( 0, 0, 0 ) );
     }
+}
+
+
+TEST_CASE( "Test get distance to point", "[distance_point]" )
+{
+    SECTION( "Simple Test" )
+    {
+        Path p = { { 0, 0, 0 }, { 5, 10, 10 }, { 10, 20, 10 } };
+        PurePursuitTest test_class( p, 5 );
+        auto dist = test_class.get_distance_to_point_test( { 6, 12 } );
+        CHECK( dist == Approx( 13.416407864 ).margin( 1e-3 ) );
+    }
+    SECTION( "Point on segment parallel to the y-axis " )
+    {
+        Path p = { { 0, 0, 0 }, { 5, 10, 10 }, { 10, 5, 10 }, { 10, 15, 10 } };
+        PurePursuitTest test_class( p, 5 );
+        auto dist = test_class.get_distance_to_point_test( { 10, 10 } );
+        CHECK( dist == Approx( 23.251407 ).margin( 1e-3 ) );
+    }
+    SECTION( "Complex Path" )
+    {
+        Path p
+            = { { 1, 3, 0 }, { 4, 1, 10 }, { 4, 4, 10 }, { 5, 15, 10 }, { 3, 10, 10 } };
+        PurePursuitTest test_class( p, 5 );
+        auto dist = test_class.get_distance_to_point_test( { 4.3, 13.25 } );
+        CHECK( dist == Approx( 19.535712 ).margin( 1e-3 ) );
+    }
+    SECTION( "Point not on path" )
+    {
+        Path p
+            = { { 1, 3, 0 }, { 4, 1, 10 }, { 4, 4, 10 }, { 5, 15, 10 }, { 3, 10, 10 } };
+        PurePursuitTest test_class( p, 5 );
+        auto dist = test_class.get_distance_to_point_test( { 100, 13.25 } );
+        auto l    = test_class.path_length_test();
```

Comment:
I added this test case, even though we expect the user to input a point on the path, we should still define the behavior for when the point does not lie on the path, in this case, make the function return -1 if the point is not on the path. 
Right now your function returns the entire length of the path when the point isn't on the path.